test/evp_test.c

```diff
@@ -7,6 +7,9 @@
  * https://www.openssl.org/source/license.html
  */
 
+/* Uses EVP_PKEY_new_CMAC_key{,_ex} */
+#define OPENSSL_SUPPRESS_DEPRECATED
```

Comment:
This won't work unless you also disable the callers when OPENSSL_NO_DEPRECATED_3_0 is set. Besides I'd just remove these test cases as we are removing the EVP_PKEY_new_CMAC_key_ex completely and the CMAC implementation is tested via the MAC API.

The EVP_PKEY_new_CMAC_key is tested also in evp_extra_test. But there it needs to be made conditional on the OPENSSL_NO_DEPRECATED_3_0 define.

Reply:
I removed some of the test and fixed `evp_extra_test` to have the right deprecation ifdefs.

Reply:
Hmmm, @t8m, not sure I agree...  we have plenty of *MAC implementation* by PKEY tests in here, so why is CMAC to be singled out specifically?  That makes zero sense.

Reply:
Because there would be no EVP_PKEY_new_CMAC_key_ex function? However I suppose we should not have EVP_PKEY_new_HMAC_key_ex either, that is for a follow up PR.

Reply:
And herein lies a problem with requiring a non-NULL libctx; we're sabotaging perfectly valid, albeit aged tests.
Do we at all run evp_test with a NULL libctx? We could make it so that these aged tests are skipped in the for non-NULL libctx. They would be a perfectly valid case for any "by PKEY" stanza.

Reply:
Unfortunately the evp_test does not work like this - it will always setup a non-null libctx. We could workaround that with the OSSL_LIB_CTX_set0_default() call though.

Reply:
Yeah well, that's the state of it *now*...  I do recall having had some heeby jeebies about the current setup, back when that hackery was started, and I guess that we now see what price we pay...

Reply:
So @t8m, @levitte do you want me to put an `OSSL_LIB_CTX_set0_default` call into `evp_test` and will that resolve the concerns?

Reply:
Yeah I think just reverting the changes in tests that were triggered by the removal of EVP_PKEY_new_CMAC_key_ex and replacing the EVP_PKEY_new_CMAC_key_ex in the tests with something like:
```
oldctx = OSSL_LIB_CTX_set0_default(libctx);
EVP_PKEY_new_CMAC_key(...);
OSSL_LIB_CTX_set0_default(oldctx);
```
should do it.

Reply:
Although... the problem is that EVP_PKEY_new_CMAC_key won't be there with no-deprecated build. So there will have to be some more conditionalizing around the EVP_PKEY_new_CMAC_key() call marking the test to be skipped in OPENSSL_NO_DEPRECATED_3_0 build.

Reply:
thanks @t8m.  Updated commit to do your set/restore the libctx.  Regular tests pass; let's see what the CI says.